diffrax/_solver/srk.py

```diff
@@ -0,0 +1,624 @@
+
+    where $W_n = W_{t_n, t_{n+1}}$ is the increment of the Brownian motion and
+    $H_n = H_{t_n, t_{n+1}}$ is its corresponding space-time Lévy Area.
+    A similar term can also be added for the space-time-time Lévy area, K.
+
+    In the special case, when the SDE has additive noise, i.e. when g is
+    independent of y (but can still depend on t), then the SDE can be written as
+    $dX_t = f(t, X_t) dt + g(t) \, dW_t$, and we can simplify the above to
+
+    $y_{n+1} = y_n + h \Big(\sum_{j=1}^s b_j k_j \Big) + g(t_n) \, (b^W
+    \, W_n + b^H \, H_n)$
+
+    $f_j = f(t_n + c_j h , z_j)$
+
+    $z_j = y_n + h \Big(\sum_{i=1}^{j-1} a_{j,i} f_i \Big) + g(t_n)
+    \, (c^W_j W_n + c^H_j H_n)$
+
+    When g depends on t, we need to add a correction term to $y_{n+1}$ of
+    the form $(g(t_{n+1}) - g(t_n)) \, (1/2 W_n - H_n)$.
+
+    The coefficients are provided in the `StochasticButcherTableau`.
+    """
+
+    term_structure: ClassVar = MultiTerm[tuple[ODETerm, AbstractTerm]]
+    interpolation_cls = LocalLinearInterpolation
+    tableau: AbstractClassVar[StochasticButcherTableau]
+
+    # Indicates the type of Levy area used by the solver.
+    # The BM must generate at least this type of Levy area, but can generate
+    # more. E.g. if the solver uses space-time Levy area, then the BM generates
+    # space-time-time Levy area as well that is fine. The other way around would
+    # not work. This is mostly an easily readable indicator so that methods know
+    # what kind of BM to use.
+    def minimal_levy_area(self):
+        if self.tableau.bK is not None:
+            return "space-time-time"
+        elif self.tableau.bH is not None:
+            return "space-time"
+        else:
+            return ""
+
+    def init(
+        self,
+        terms: term_structure,
+        t0: RealScalarLike,
+        t1: RealScalarLike,
+        y0: Y,
+        args: PyTree,
+    ) -> _SolverState:
+        # Check that the diffusion has the correct Levy area
+        _, diffusion = terms.terms
+
+        stla = self.tableau.bH is not None
+        sttla = self.tableau.bK is not None
+
+        is_bm = lambda x: isinstance(x, AbstractBrownianPath)
+        leaves = jtu.tree_leaves(diffusion, is_leaf=is_bm)
+        paths = [x for x in leaves if is_bm(x)]
+        for path in paths:
+            if sttla:
+                if not path.levy_area == "space-time-time":
+                    raise ValueError(
+                        "The Brownian path controlling the diffusion "
+                        "should be initialised with `levy_area='space-time-time'`"
+                    )
+            elif stla:
+                if path.levy_area not in ["space-time", "space-time-time"]:
+                    raise ValueError(
+                        "The Brownian path controlling the diffusion "
+                        "should be initialised with `levy_area='space-time'`"
+                        "or `levy_area='space-time-time'`"
+                    )
```

Comment:
By the way, can you cast your eye at #364? The hope is to try and encode the return type of the control into the term, i.e.
```python
class AbstractTerm(Generic[T], ...):
    def contr(...) -> T:
        ...
```
and in doing so then hopefully we should be able to explicitly declare ` term_structure: ClassVar = MultiTerm[tuple[ODETerm, AbstractTerm[AbstractLevyArea]]]`, and ideally have that be checked by `diffeqsolve`, without needing to check it in ad-hoc fashion here.

Reply:
(I hope this also answers the question you had earlier about "determining compatibility".)

Reply:
Sounds great! I am already using the return type to check if the correct BM was supplied (i.e. I check if it is a `LevyVal` and has the required Levy areas). That still doesn't differentiate between UnsafeBrownianPath and VBT, though. How do you propose we check that?

Reply:
So I don't think there *should* be a way to distinguish between different control paths themselves. In some sense this is kind of the point of the Term abstraction, e.g. that we can write down the same Euler solver regardless of whether the control is time or a Brownian motion.